src/PantheonTerminalWindow.vala

```diff
@@ -49,39 +49,69 @@ namespace PantheonTerminal {
-                <menuitem name="Search" action="Search"/>
-                <menuitem name="About" action="About"/>
-
-                <menuitem name="NextTab" action="NextTab"/>
-                <menuitem name="PreviousTab" action="PreviousTab"/>
-
-                <menuitem name="ZoomIn" action="ZoomIn"/>
-                <menuitem name="ZoomOut" action="ZoomOut"/>
-
-                <menuitem name="Fullscreen" action="Fullscreen"/>
-            </popup>
-
-            <popup name="AppMenu">
-                <menuitem name="Copy" action="Copy"/>
-                <menuitem name="Paste" action="Paste"/>
-                <menuitem name="Select All" action="Select All"/>
-                <menuitem name="Search" action="Search"/>
-                <menuitem name="Show in File Browser" action="Show in File Browser"/>
-            </popup>
-            </ui>
-        """;
+        public SimpleActionGroup actions { get; construct; }
+
+        public const string ACTION_PREFIX = "win.";
+        public const string ACTION_CLOSE_TAB = "action_close_tab";
+        public const string ACTION_COPY = "action_copy";
+        public const string ACTION_FULLSCREEN = "action_fullscreen";
+        public const string ACTION_NEW_TAB = "action_new_tab";
+        public const string ACTION_NEW_WINDOW = "action_new_window";
+        public const string ACTION_NEXT_TAB = "action_next_tab";
+        public const string ACTION_SHOW_IN_BROWSER = "action_open_in_files";
+        public const string ACTION_PASTE = "action_paste";
+        public const string ACTION_PREVIOUS_TAB = "action_previous_tab";
+        public const string ACTION_QUIT = "action_quit";
+        public const string ACTION_SEARCH = "action_search";
+        public const string ACTION_SELECT_ALL = "action_select_all";
+        public const string ACTION_ZOOM_IN_FONT = "action_zoom_in_font";
+        public const string ACTION_ZOOM_OUT_FONT = "action_zoom_out_font";
+
+        public static Gee.MultiMap<string, string> action_accelerators = new Gee.HashMultiMap<string, string> ();
+
+        private const ActionEntry[] action_entries = {
+            { ACTION_CLOSE_TAB, action_close_tab },
+            { ACTION_COPY, action_copy },
+            { ACTION_FULLSCREEN, action_fullscreen },
+            { ACTION_NEW_TAB, action_new_tab },
+            { ACTION_NEW_WINDOW, action_new_window },
+            { ACTION_NEXT_TAB, action_next_tab },
+            { ACTION_SHOW_IN_BROWSER, action_open_in_files },
+            { ACTION_PASTE, action_paste },
+            { ACTION_PREVIOUS_TAB, action_previous_tab },
+            { ACTION_QUIT, action_quit },
+            { ACTION_SEARCH, action_search },
+            { ACTION_SELECT_ALL, action_select_all },
+            { ACTION_ZOOM_IN_FONT, action_zoom_in_font },
+            { ACTION_ZOOM_IN_FONT, action_zoom_out_font }
+        };
+
+        static construct {
+            action_accelerators.set (ACTION_CLOSE_TAB, "<Control><Shift>w");
```

Comment:
You can use a simpler syntax for this:
`action_accelerators[ACTION_CLOSE_TAB] = "<Control><Shift>w";`